Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/view/step_list/StepListFragment.java

```diff
@@ -31,15 +34,39 @@
 
     StepListRecyclerViewAdapter.StepItemClickListener stepItemClickListener =
             new StepListRecyclerViewAdapter.StepItemClickListener() {
+
+                private boolean removedStep = false;
                 @Override
                 public void onRemoveStepClick(long itemId) {
                     stepTemplateRepository.delete(itemId);
                     stepListRecyclerViewAdapter
                             .setStepItemListItems(stepTemplateRepository.getAll(task_id));
+                    toastUserNotifier.displayNotifications(
+                            R.string.step_removed_message,
+                            getActivity().getApplicationContext());
+                    removedStep = true;
+                }
+
+                @Override
+                public void onMoveItem() {
+                    Boolean reordered = false;
+                    for(int i = 0; i < stepListRecyclerViewAdapter.getItemCount(); i++){
```

Comment:
In my opinion, It would be nicer if we could change **i** to something more meaningful like **positionOnList** or **currentOrder**.  It would show, that we are changing step order here. But it could be changed later.